examples/shell/shell_common/cmd_ping.cpp

```diff
@@ -335,7 +327,7 @@ void StartPinging(streamer_t * stream, char * destination)
         // Wait for response until the Echo interval.
         while (!EchoIntervalExpired())
         {
-            DriveIO(stream);
+            sleep(1);
```

Comment:
Can't we do better than sleep here?

Reply:
After we remove DriveIO, the main thread is not blocking on any IO, we need to make sure we are not abusing CPU cycles here. 

Using sleep is simplest way to achieve that. Otherwise we need to use some thread constructs which suspect the current thread and wake it up as soon as receiving any response.

Reply:
> Otherwise we need to use some thread constructs which suspect the current thread and wake it up as soon as receiving any response.

Exactly. Let's do that.

Reply:
@mspang 

It looks nrfconnect does not support C++11 concurrency constructs, those constructs works fine on POSIX platforms.

/var/chip/examples/shell/shell_common/cmd_ping.cpp:143:6: error: 'condition_variable' in namespace 'std' does not name a type
  143 | std::condition_variable gCond;
      |      ^~~~~~~~~~~~~~~~~~
/var/chip/examples/shell/shell_common/cmd_ping.cpp:248:10: error: 'mutex' is not a member of 'std'
  248 |     std::mutex mtx;
      |          ^~~~~

Reply:
@Damian-Nordic

Reply:
I see the C++14 is enabled for Nordic board, CONFIG_STD_CPP14=y, but I know some advanced features are usually   disabled in highly embedded platforms.

Reply:
Created https://github.com/project-chip/connectedhomeip/issues/5496 to track this issue on Nordic. Temporally use sleep to force context switch now.

Reply:
I think we may need to introduce some abstraction to accommodate platforms whose libc doesn't provide `std ::condition_variable`. I'll take a look at the issue and propose something.